syntax = "proto3";

package unary;

service Unary {
    rpc Start(EmptyParam) returns (Response) {}
    rpc Stop(EmptyParam) returns (Response) {}
    rpc Pause(EmptyParam) returns (Response) {}
    rpc CreateBody(CreateBodyParam) returns (Response) {}
    rpc UpdateBody(UpdateBodyParam) returns (Response) {}
    rpc DeleteBody(DeleteBodyParam) returns (Response) {}
}

message CreateBodyParam {
    int32 id = 1;
    float p_x = 2;
    float p_y = 3;
    float v_x = 4;
    float v_y = 5;
}

message DeleteBodyParam {
    int32 id = 1;
}

message UpdateBodyParam {
    int32 id = 1;
    optional float p_x = 2;
    optional float p_y = 3;
    optional float v_x = 4;
    optional float v_y = 5;
}

message EmptyParam {}

message Response {
    bool received = 1;
}